### ISSUE TITLE
File Menu: Open Level and Save Level As dialogs are not properly recognized as modals

### ISSUE THREAD

**Issue by jckand-amzn** [o3de/o3de#6605]
**Describe the bug**
Open Level and Save Level As dialogs are not properly recognized as modals. This causes the dialogs to block script execution in automated tests using autotest_mode.

**Steps to reproduce**
1. Uncomment lines 38 and 41 from AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Menus_FileMenuOptions.py
2. Launch the Editor with the AutomatedTesting project in --autotest_mode
3. In the Console, type:

```
pyRunFile <path to AutomatedTesting>/Gem/PythonTests/editor/EditorScripts/Menus_FileMenuOptions.py
```

**Expected behavior**
Script completes without issue, dismissing dialogs as needed.

**Actual behavior**
Open Level dialog never dismisses, blocking further script execution.

**Found in Branch**
development@24a3014e065

**Comment by jckand-amzn** [comment]
.show() and .adjustSize() calls before exec for these dialogs appears to be the culprit. If commented out, the test completes without issue.

**Comment by jjjoness** [comment]
I am looking at this, but lines 38 and 41 are not commented:-

**Comment by jckand-amzn** [comment]
Heya, @jjjoness it is the correct file. The changes with the commenting are still in PR, but should be in today. The same issue should happen on development as is, however, you just need to comment out the ```@pytest.mark.skip(reason="Times out due to dialogs failing to dismiss: LYN-4208")``` line from the test runner at AutomatedTesting\Gem\PythonTests\editor\TestSuite_Main_Optimized.py.

**Comment by jjjoness** [comment]
@jckand-amzn Thanks for that, I'll get back on it tomorrow.

**Comment by jjjoness** [comment]
This should have ben closed by #7072. Closing now.
